src/coordinator/functions/logical/or.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package logical
+
+import (
+	"fmt"
+
+	"github.com/m3db/m3db/src/coordinator/block"
+	"github.com/m3db/m3db/src/coordinator/executor/transform"
+	"github.com/m3db/m3db/src/coordinator/parser"
+)
+
+//TODO: generalize logical functions?
+const (
+	// OrType uses all values from left hand side, and appends values from the right hand side which do
+	// not have corresponding tags on the right
+	OrType = "or"
+)
+
+// NewOrOp creates a new Or operation
+func NewOrOp(lNode parser.NodeID, rNode parser.NodeID, matching *VectorMatching) BaseOp {
+	return BaseOp{
+		OperatorType: OrType,
+		LNode:        lNode,
+		RNode:        rNode,
+		Matching:     matching,
+		ProcessorFn:  NewOrNode,
+	}
+}
+
+// OrNode is a node for And operation
+type OrNode struct {
+	op         BaseOp
+	controller *transform.Controller
+}
+
+// NewOrNode creates a new OrNode
+func NewOrNode(op BaseOp, controller *transform.Controller) Processor {
+	return &OrNode{
+		op:         op,
+		controller: controller,
+	}
+}
+
+// Process processes two logical blocks, performing Or operation on them
+func (c *OrNode) Process(lhs, rhs block.Block) (block.Block, error) {
+	builder, err := c.controller.BlockBuilder(lhs.Meta(), lhs.SeriesMeta())
```

Comment:
this probably needs to change since you can't simply use lhs.SeriesMeta and lhs.Meta

Reply:
Oops, didn't update